client/extensions/zoninator/state/data-layer/zones/test/index.js

```diff
@@ -4,29 +4,29 @@
 import { expect } from 'chai';
 import sinon from 'sinon';
 import { translate } from 'i18n-calypso';
-import {
-	startSubmit as startSave,
-	stopSubmit as stopSave,
-} from 'redux-form';
+import { startSubmit, stopSubmit } from 'redux-form';
 
 /**
  * Internal dependencies
  */
 import { http } from 'state/data-layer/wpcom-http/actions';
 import { errorNotice, removeNotice, successNotice } from 'state/notices/actions';
 import {
-	announceFailure,
+	announceDeleteFailure,
```

Comment:
Should there be tests for `announceZoneUpdated` and `announceZoneDeleted`?

Reply:
The issue with those two is that they call `page` which makes the tests fail.
I haven't figured out a good way to mock it yet.
It's the same issue with `settingsPath` - it's originally exported in `zoninator/app/util.js` but the tests can't compile if I import it here.

Reply:
I would have thought you could use `sinon.stub` for that, but perhaps not - https://github.com/sinonjs/sinon/issues/664. They mention using `proxyquire` which may be worth investigating, but not something I'd spend a ton of time looking into.

Reply:
I think the correct solution here could be to create a global `PAGE_REDIRECT` action that calls `page()` in its handler and dispatch that whenever necessary.  
That would make the handlers testable again.

I found `ROUTE_SET` but that only seems to affect the url without actually changing the view.